static/app/types/profiling/trace.ts

```diff
@@ -0,0 +1,40 @@
+type Annotation = {
+  key: string;
+  values: string[];
+};
+
+export type Trace = {
+  app_version: string;
+  device_class: string;
+  device_locale: string;
+  device_manufacturer: string;
+  device_model: string;
+  id: string;
+  interaction_name: string;
+  start_time_unix: number;
+  trace_duration_ms: number;
+  backtrace_available?: boolean;
+  error_code?: number;
+  error_code_name?: string;
+  error_description?: string;
+  failed?: boolean;
+  span_annotations?: Annotation[];
```

Comment:
For extra safety (and because we tend to do some operations on these spans to display them), wdyt about using a ReadonlyArray<Annotation> here? (same for other array values)